geode-assembly/src/test/java/org/apache/geode/session/tests/TomcatSessionBackwardsCompatibilityTest.java

```diff
@@ -0,0 +1,254 @@
+    CommandStringBuilder locStarter = new CommandStringBuilder(CliStrings.START_LOCATOR);
+    locStarter.addOption(CliStrings.START_LOCATOR__MEMBER_NAME, "loc");
+    locStarter.addOption(CliStrings.START_LOCATOR__CLASSPATH, classPath);
+    locStarter.addOption(CliStrings.START_LOCATOR__PORT, Integer.toString(port));
+    gfsh.executeAndVerifyCommand(locStarter.toString());
+
+  }
+
+  @Before
+  public void setup() throws Exception {
+    tomcat7079AndOldModules = new TomcatInstall(TomcatInstall.TomcatVersion.TOMCAT779,
+        ContainerInstall.ConnectionType.CLIENT_SERVER,
+        ContainerInstall.DEFAULT_INSTALL_DIR + "Tomcat7079AndOldModules",
+        oldModules.getAbsolutePath(), oldBuild.getAbsolutePath() + "/lib");
+
+    tomcat7079AndCurrentModules = new TomcatInstall(TomcatInstall.TomcatVersion.TOMCAT779,
+        ContainerInstall.ConnectionType.CLIENT_SERVER,
+        ContainerInstall.DEFAULT_INSTALL_DIR + "Tomcat7079AndCurrentModules");
+
+    tomcat8AndOldModules = new TomcatInstall(TomcatInstall.TomcatVersion.TOMCAT8,
+        ContainerInstall.ConnectionType.CLIENT_SERVER,
+        ContainerInstall.DEFAULT_INSTALL_DIR + "Tomcat8AndOldModules", oldModules.getAbsolutePath(),
+        oldBuild.getAbsolutePath() + "/lib");
+
+    tomcat8AndCurrentModules = new TomcatInstall(TomcatInstall.TomcatVersion.TOMCAT8,
+        ContainerInstall.ConnectionType.CLIENT_SERVER,
+        ContainerInstall.DEFAULT_INSTALL_DIR + "Tomcat8AndCurrentModules");
+
+    classPathTomcat7079 = tomcat7079AndCurrentModules.getHome() + "/lib/*" + File.pathSeparator
+        + tomcat7079AndCurrentModules.getHome() + "/bin/*";
+    classPathTomcat8 = tomcat8AndCurrentModules.getHome() + "/lib/*" + File.pathSeparator
+        + tomcat8AndCurrentModules.getHome() + "/bin/*";
+
+    // Get available port for the locator
+    locatorPort = AvailablePortHelper.getRandomAvailableTCPPort();
+
+    tomcat7079AndOldModules.setDefaultLocator("localhost", locatorPort);
+    tomcat7079AndCurrentModules.setDefaultLocator("localhost", locatorPort);
+
+    tomcat8AndOldModules.setDefaultLocator("localhost", locatorPort);
+    tomcat8AndCurrentModules.setDefaultLocator("localhost", locatorPort);
+
+    client = new Client();
+    manager = new ContainerManager();
+    // Due to parameterization of the test name, the URI would be malformed. Instead, it strips off
+    // the [] symbols
+    manager.setTestName(testName.getMethodName().replace("[", "").replace("]", ""));
+  }
+
+  private void startClusterWithTomcat(String tomcatClassPath) throws Exception {
+    startLocator("loc", tomcatClassPath, locatorPort);
+    startServer("server", tomcatClassPath, locatorPort);
+  }
+
+  /**
+   * Stops all containers that were previously started and cleans up their configurations
+   */
+  @After
+  public void stop() throws Exception {
+    manager.stopAllActiveContainers();
+    manager.cleanUp();
+
+    tomcat8AndCurrentModules.clearPreviousRuns();
```

Comment:
Change clearPreviousRuns to remove only previous runs of current version being added